src/main/java/org/template/Algorithm.java

```diff
@@ -850,29 +850,95 @@ public PredictedResult predict(NullModel model, Query query) {
       obj.addProperty("boost", bc.boost);
       shouldFields.add(obj);
     }
-
-    String shouldScore =
-        "{\n" +
-            "   \"constant_score\": {\n" +
-            "  \"filter\": {\n" +
-            "  \"match_all\": {}\n" +
-            "},\n" +
-            "  \"boost\": 0\n" +
-            "  }\n" +
-            "}";
-    shouldFields.add(
-        new JsonParser().parse(shouldScore).getAsJsonObject()
-    );
-
-    return shouldFields;
-  }
-
-  private List<JsonElement> buildQueryMustNot(Query query, List<Event> boostable) {
-    return null;
-  }
-
-  private List<JsonElement> buildQuerySort() {
-    return null;
+      
+      String shouldScore =
+      "{\n" +
+      "   \"constant_score\": {\n" +
+      "  \"filter\": {\n" +
+      "  \"match_all\": {}\n" +
+      "},\n" +
+      "  \"boost\": 0\n" +
+      "  }\n" +
+      "}";
+      shouldFields.add(
+                       new JsonParser().parse(shouldScore).getAsJsonObject()
+                       );
+      
+      return shouldFields;
   }
 
+    /** Build sort query part */
+    private List<JsonElement> buildQuerySort(){
+        List<JsonElement> Seq = new ArrayList <JsonElement>();
+        if (recsModel == RecsModel.All || recsModel == RecsModel.BF){
+            Gson gson = new Gson();
+            List <JsonElement> sortByScore = new ArrayList <JsonElement>();
+            List <JsonElement> sortByRanks = new ArrayList <JsonElement>();
+            sortByScore.add(new JsonParser().parse("{\"_score\": {\"order\": \"desc\"}}"));
+            for(String fieldName:rankingFieldNames){
+                sortByRanks.add(new JsonParser().parse("{ \"$fieldName\": { \"unmapped_type\": \"double\", \"order\": \"desc\" } }"));
+            }
+            sortByScore.addAll(sortByRanks);
+            return sortByScore;
+        }
+        else{
+            return Seq;
```

Comment:
[style] instead of declaring a local var Seq, can just create an empty list here and return